nautobot/extras/api/serializers.py

```diff
@@ -563,6 +564,9 @@ def validate(self, data):
 
         return super().validate(data)
 
+    def get_task_queues(self, obj):
+        return list(obj.task_queues) if obj is not None else obj
```

Comment:
`JobSerializer(...).data` returns `{..."task_queues": <RestrictedQuerySet ...>}` which is not JSON serialisable, hence the change

Reply:
Do we just need to use the right JSON encoder class? `DjangoJSONEncoder` handles UUIDs and datetimes correctly. Just need to understand why this is working fine for the REST API as-is, but causing issues only for event publishing.

Reply:
Using DjangoJSONEncoder did not resolve the issue. The Django REST Framework (DRF) addresses this through rest_framework.renderers.JSONRenderer, and in Nautobot, this is handled by nautobot.core.api.renderers.NautobotJSONRenderer. By leveraging NautobotKombuJSONEncoder as the cls argument in json.dumps, we can properly serialize objects like the RestrictedQuerySet found in "task_queues" into valid JSON, e.g., json.dumps(data, cls=NautobotKombuJSONEncoder).

One possible solution is to apply this fix globally by ensuring that json.dumps is used in the Broker.publish method. However, I’m uncertain if brokers would prefer to handle Python objects directly and manage the json.dumps process themselves, or if applying json.dumps beforehand would be sufficient.

Reply:
As discussed in standup, let's go with pre-dumping the payload to a JSON string so that each individual broker doesn't need to use the right encoder/renderer.